waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/SimpleFederationStatusService.java

```diff
@@ -44,17 +46,13 @@ private HiveConf newHiveConf(String metaStoreUris) {
    *         returns {@code MetaStoreStatus.UNAVAILABLE}
    */
   @Override
-  public MetaStoreStatus checkStatus(String metaStoreUris) {
-    HiveConf hiveConf = newHiveConf(metaStoreUris);
-    HiveMetaStoreClient client = null;
-    try {
-      client = new HiveMetaStoreClient(hiveConf);
+  public MetaStoreStatus checkStatus(AbstractMetaStore abstractMetaStore) {
+    try (CloseableThriftHiveMetastoreIface client = metaStoreClientFactory.newInstance(abstractMetaStore)) {
+      if (!client.isOpen()) {
+        return MetaStoreStatus.UNAVAILABLE;
+      }
```

Comment:
Where does the closing of the client happen now?

Reply:
Try with resources, java does it for me \o/

Reply:
Aah, of course!